### PR TITLE
Display the downloaded size in QgsFileDownloaderDialog

### PR DESCRIPTION
**Pull request by Gustry**
## Description

Display the downloaded size in QgsFileDownloaderDialog

@elpaso Can I have your review? Thanks

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Review by elpaso — APPROVED** [review]
LGTM

**Comment by Gustry** [comment]
Can it be merged in 3.0? It's only a cut/paste from `humanSize`

File: src/core/qgsfileutils.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsFileUtils
+{
+  public:
+
+    /**
+     * Return the human size from bytes
+     */
+    static QString humanSize( qint64 bytes );
```
Comment (m-kuhn) [inline src/core/qgsfileutils.h]: I assume the human size varies between ~1.5 m and 2m in general.

Maybe you meant human readable size and a method name like `representFileSize` would be better suited? ;)

File: src/core/qgsfileutils.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  QStringListIterator i( list );
+  QString unit( "bytes" );
+
+  while ( bytes >= 1024.0 && i.hasNext() )
+  {
+    unit = i.next();
+    bytes /= 1024.0;
+  }
+  return QString( "%1 %2" ).arg( QString::number( bytes ) ).arg( unit );
```
Comment (m-kuhn) [inline src/core/qgsfileutils.cpp]: `QStringLiteral(...).arg`

File: src/core/qgsfileutils.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include "qgsfileutils.h"
+
+QString QgsFileUtils::humanSize( qint64 bytes )
+{
+  QStringList list;
+  list << "KB" << "MB" << "GB" << "TB";
```
Comment (m-kuhn) [inline src/core/qgsfileutils.cpp]: Does this possibly need to be localized? Thinking mostly about languages with non-latin character sets.

File: src/gui/qgsfiledownloaderdialog.cpp [hunk 4]
```diff
@@ -46,5 +48,6 @@ void QgsFileDownloaderDialog::onDownloadProgress( qint64 bytesReceived, qint64 b
 {
   setMaximum( bytesTotal );
   setValue( bytesReceived );
+  setLabelText( tr( "Downloading %1 of %2 %3." ).arg( QgsFileUtils::humanSize( bytesReceived ) ).arg( QgsFileUtils::humanSize( bytesTotal ) ).arg( mOutputFileName ) );
```
Comment (m-kuhn) [inline src/gui/qgsfiledownloaderdialog.cpp]: A single `.arg()` should be more efficient than the 3 chained ones here.

**Comment by m-kuhn** [comment]
> Can it be merged in 3.0? It's only a cut/paste of humanSize

That's fine with me, it's quite a handy thing to have.

File: src/core/qgsfileutils.h (hunk 1 above)
Comment (elpaso) [reply src/core/qgsfileutils.h]: `ls` and many other standard unix commands such as `free` have a `-h` argument which stands for `human readable`. Maybe we can call it `humanReadableSize`.

Comment (m-kuhn) [reply src/core/qgsfileutils.h]: my suggestion was based on `QgsFieldFormatter::representValue()` and friends.

File: src/core/qgsfileutils.cpp [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#include "qgsfileutils.h"
+#include <QObject>
+
+QString QgsFileUtils::humanReadableSize( qint64 bytes )
+{
+  QStringList list;
+  list << QObject::tr( "KB" ) << QObject::tr( "MB" ) << QObject::tr( "GB" ) << QObject::tr( "TB" );
```
Comment (timlinux) [inline src/core/qgsfileutils.cpp]: We seem to be missing yottabytes from this list :-P

**Comment by Gustry** [comment]
Is-it all fine like this?
I rolled back to Matthias suggestion

**Comment by m-kuhn** [comment]
Looks great to me, thanks a log @Gustry

**Comment by m-kuhn** [comment]
@timlinux can we proceed without yottabytes or do you need that for local internet speed per second?

**Pull request merged**